v03_pipeline/lib/misc/validation_test.py

```diff
@@ -29,6 +32,57 @@ def _mt_from_contigs(contigs):
 
 
 class ValidationTest(unittest.TestCase):
+    def test_validate_imputed_sex_ploidy(self) -> None:
+        sex_check_ht = hl.read_table(TEST_SEX_CHECK_1)
+        mt = hl.MatrixTable.from_parts(
+            rows={
+                'locus': [
+                    hl.Locus(
+                        contig='chr1',
+                        position=1,
+                        reference_genome='GRCh38',
+                    ),
+                ],
+            },
+            cols={'s': ['HG00731_1', 'HG00732_1']},
+            entries={
+                'GT': [
+                    [
+                        hl.Call(alleles=[0, 0], phased=False),
+                        hl.Call(alleles=[0], phased=False),
+                    ],
+                ],
+            },
+        ).key_rows_by('locus')
+        validate_imputed_sex_ploidy(mt, sex_check_ht)
+        mt = hl.MatrixTable.from_parts(
+            rows={
+                'locus': [
+                    hl.Locus(
+                        contig='chr1',
+                        position=1,
+                        reference_genome='GRCh38',
+                    ),
+                ],
+            },
+            cols={'s': ['HG00731_1', 'HG00732_1']},
+            entries={
+                'GT': [
+                    [
+                        hl.Call(alleles=[0], phased=False),
+                        hl.Call(alleles=[0], phased=False),
+                    ],
+                ],
+            },
+        ).key_rows_by('locus')
+        self.assertRaisesRegex(
+            SeqrValidationError,
+            '50.00% of samples have misaligned ploidy',
```

Comment:
hmm, what's in the validation ht? both would be flagged as having haploid calls, but I'm assuming _1 is predicted female and _2 is predicted male?

Reply:
I've got a print out [here](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/tasks/write_remapped_and_subsetted_callset_test.py#L26-L47) but yes `HG00731_1` is female and `HG00732_1` is male.

Reply:
yeah that tracks 

would we want a test for unknown?